File: api-java/samples/server/trinsic/Direct.java
package id.trinsic;
import java.util.UUID;
import id.trinsic.api.SessionsApi;
import id.trinsic.api.models.*;
import io.javalin.Javalin;

import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.time.Instant;
import java.util.*;
import java.util.stream.Collectors;
import java.util.UUID;

public class Direct {
    public static void DirectRoutes(Javalin app, SessionsApi session, UUID verificationProfileId){

        app.get("/direct-popup", ctx -> {
            String query = ctx.queryString(); // grabs everything after '?'
            String target = "/direct-popup.html" + (query != null ? "?" + query : "");
            ctx.redirect(target);
        });

        app.post("/create-direct-session/{provider}", ctx -> {
            String provider = ctx.pathParam("provider");
            String redirectUrl = ctx.queryParam("redirectUrl");
            boolean fallbackToTrinsicUI = "true".equals(ctx.queryParam("fallbackToTrinsicUI"));

            // Build request
            CreateDirectProviderSessionRequest req = new CreateDirectProviderSessionRequest();
            req.setProvider(provider);
            req.setRedirectUrl(redirectUrl);
            req.setFallbackToHostedUI(fallbackToTrinsicUI);
            req.setVerificationProfileId(verificationProfileId);

            String capabilitiesParam = ctx.queryParam("capabilities");
            List<IntegrationCapability> capabilities = capabilitiesParam != null
                    ? Arrays.stream(capabilitiesParam.split(","))
                    .map(String::trim)
                    .map(IntegrationCapability::fromValue)
                    .collect(Collectors.toList())
                    : List.of();

            req.setCapabilities(capabilities);

            // Call API
            CreateDirectProviderSessionResponse result =
                    session.createDirectProviderSession(req);

            ctx.json(result);
        });

        app.post("/refresh-content/{sessionId}", ctx -> {
            String sessionId = ctx.pathParam("sessionId");

            Map<String, Object> body = ctx.bodyAsClass(Map.class);
            String resultsAccessKey = (String) body.get("resultsAccessKey");

            RefreshStepContentRequest request = new RefreshStepContentRequest();
            request.setResultsAccessKey(resultsAccessKey);

            RefreshStepContentResponse result =
                    session.refreshStepContent(UUID.fromString(sessionId), request);

            ctx.json(result);
        });

        app.post("/poll-results/{sessionId}", ctx -> {
            String sessionId = ctx.pathParam("sessionId");

            Map<String, Object> body = ctx.bodyAsClass(Map.class);
            String resultsAccessKey = (String) body.get("resultsAccessKey");

            GetSessionResultRequest request = new GetSessionResultRequest();
            request.setResultsAccessKey(resultsAccessKey);

            GetSessionResultResponse result =
                    session.getSessionResult(UUID.fromString(sessionId), request);

            ctx.json(result);
        });
    }
}

File: api-php/samples/server/src/direct.php
<?php

use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use Trinsic\Api\Model\CreateDirectProviderSessionRequest as CreateDirectProviderSessionRequest;
use Trinsic\Api\Model\CreateDirectProviderSessionResponse as CreateDirectProviderSessionResponse;
use Trinsic\Api\Model\GetSessionResultRequest as GetSessionResultRequest;
use Trinsic\Api\Model\RefreshStepContentRequest as RefreshStepContentRequest;
use Trinsic\Api\ApiException as ApiException;

return function ($app, $sessions) {

    $app->post("/create-direct-session/{provider}", function (Request $request, Response $response, $args) use ($sessions) {
        $provider = $args['provider'];
        $queryParams = $request->getQueryParams();
        $redirectUrl = $queryParams['redirectUrl'] ?? null;
        $fallbackToTrinsicUI = ($queryParams['fallbackToTrinsicUI'] ?? 'false') === 'true';
        $capabilities = isset($queryParams['capabilities']) ? explode(",", $queryParams['capabilities']) : [];

        $req = new CreateDirectProviderSessionRequest();
        $req->setRedirectUrl($redirectUrl);
        $req->setProvider($provider);
        $verificationProfileId = getenv('TRINSIC_VERIFICATION_PROFILE_ID');
        $req->setVerificationProfileId($verificationProfileId);
        $req->setCapabilities($capabilities);
        $req->setFallbackToHostedUi($fallbackToTrinsicUI);

        $result = $sessions->createDirectProviderSession($req);

        $response->getBody()->write(json_encode($result));
        return $response->withHeader('Content-Type', 'application/json');
        
    });
    
    $app->post("/refresh-content/{sessionId}", function (Request $request, Response $response, $args) use ($sessions) {
        $sessionId = $args['sessionId'];
        $body = json_decode($request->getBody()->getContents(), true);
        $resultsAccessKey = $body['resultsAccessKey'] ?? null;
        $req = new RefreshStepContentRequest();
        $req->setResultsAccessKey($resultsAccessKey);
        $result = $sessions->refreshStepContent($sessionId, $req);
    
        $response->getBody()->write(json_encode($result));
        return $response->withHeader('Content-Type', 'application/json');
    });
    
    $app->post("/poll-results/{sessionId}", function (Request $request, Response $response, $args) use ($sessions) {
        $sessionId = $args['sessionId'];
        $body = json_decode($request->getBody()->getContents(), true);
        $resultsAccessKey = $body['resultsAccessKey'] ?? null;
    
        $req = new GetSessionResultRequest();
        $req->setResultsAccessKey($resultsAccessKey);
    
        $result = $sessions->getSessionResult($sessionId, $req);
    
        $response->getBody()->write(json_encode($result));
        return $response->withHeader('Content-Type', 'application/json');
    });

};